pdq/CAM.go

```diff
@@ -0,0 +1,19 @@
+package pdq
+
+import (
+	"fmt"
+)
+
+// https://github.com/4rchr4y/go-code-metric#cohesion-among-methods-in-class-cam
+
+func CalcCAM(sharedPairs int, totalPossiblePairs int) (float64, error) {
+	if sharedPairs < 0 || totalPossiblePairs < 0 {
+		return 0, fmt.Errorf("shared pairs and total possible number of methods pairs cannot be negative, but got %d, %d", sharedPairs, totalPossiblePairs)
+	}
+
+	if sharedPairs == 0 || totalPossiblePairs == 0 {
+		return 0, nil
+	}
```

Comment:
same situation with clarification of behavior at zero values ​​as [here](https://github.com/4rchr4y/go-code-metric/pull/8/files#r1393626932)